near-plugins/tests/full_access_key_fallback.rs

```diff
@@ -0,0 +1,171 @@
+// Using `pub` to avoid invalid `dead_code` warnings, see
+// https://users.rust-lang.org/t/invalid-dead-code-warning-for-submodule-in-integration-test/80259
+pub mod common;
+
+use anyhow::Ok;
+use common::full_access_key_fallback_contract::FullAccessKeyFallbackContract;
+use common::utils::{assert_only_owner_permission_failure, assert_success_with_unit_return};
+use near_sdk::serde::Deserialize;
+use near_sdk::serde_json::{from_value, json};
+use std::iter;
+use std::path::Path;
+use workspaces::network::Sandbox;
+use workspaces::types::{AccessKeyPermission, PublicKey};
+use workspaces::{Account, AccountId, Contract, Worker};
+
+const PROJECT_PATH: &str = "./tests/contracts/full_access_key_fallback";
+
+/// Returns a new PublicKey that can be used in tests.
+///
+/// It returns a `near_sdk::PublicKey` since that's the type required for
+/// `FullAccessKeyFallback::attach_full_access_key`.
+fn new_public_key() -> near_sdk::PublicKey {
+    "ed25519:6E8sCci9badyRkXb3JoRpBj5p8C6Tw41ELDZoiihKEtp"
+        .parse()
+        .unwrap()
+}
+
+/// Converts a `near_sdk::PublicKey` to a `workspaces::types::PublicKey`.
+fn pk_sdk_to_workspaces(public_key: near_sdk::PublicKey) -> PublicKey {
```

Comment:
The implementation of this function is pretty hilarious. Serializing to JSON and back as a type conversion seems quite inefficient. Is there not another way to convert between these types? If not it's maybe worth raising as an issue in the workspaces repo.

Reply:
`near-sdk` has `PublicKey` but no `SecretKey` type, so I think the conversion has to be via `PublicKey`. Looking at the [workspaces::types::PublicKey docs](https://docs.rs/workspaces/0.7.0/workspaces/types/struct.PublicKey.html#impl-Into%3CU%3E-for-PublicKey), I only see `Deserialize<'de>` for generating a `workspaces::types::PublicKey` from data.

Since this function is only available in tests, I wasn’t too concerned about the efficiency of the conversion. Though it’s definitely awkward. I’ve opened [this issue](https://github.com/near/workspaces-rs/issues/262) and left a comment in the code.

---

Perhaps there’s a more efficient way to do this conversion via `serde`, though not sure if it’s worth digging into that.